sys/fmt/fmt.c

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ * @ingroup sys_fmt
+ * @{
+ * @file
+ * @brief String formatting library implementation
+ * @author Kaspar Schleiser <kaspar@schleiser.de>
+ * @}
+ */
+
+#include <assert.h>
+#include <stdarg.h>
+#include <stdint.h>
+#include <unistd.h>
+
+#include "div.h"
+#include "fmt.h"
+
+static const char _hex_chars[16] = "0123456789ABCDEF";
+
+static inline int _is_digit(char c)
+{
+    return (c >= '0' && c <= '9');
+}
+
+size_t fmt_byte_hex(char *out, uint8_t byte)
+{
+    if (out) {
+        *out++ = _hex_chars[byte >> 4];
+        *out = _hex_chars[byte & ~0xF0];
+    }
+    return 2;
+}
+
+size_t fmt_strlen(const char *str)
+{
+    const char *tmp = str;
+    while(*tmp) {
+        tmp++;
+    }
+    return (tmp - str);
+}
+
+size_t fmt_str(char *out, const char *str)
+{
+    char c;
+    int len = 0;
+    if (!out) {
+        len = fmt_strlen(str);
```

Comment:
All functions always return the needed bytes. If out is != NULL, the bytes actually get written. Or is there a mismatch with the Doxygen?